testing/src/test/java/org/motechproject/nms/testing/it/api/UserControllerBundleIT.java

```diff
@@ -453,50 +346,29 @@ private void createFlwWithLanguageLocationCodeNotInWhitelist() {
 
     private void createCircleWithLanguage() {
 
-        Language papiamento = new Language("99", "Papiamento");
-        languageDataService.create(papiamento);
+        // Let's create a pretend circle with two states
 
-        District district = new District();
-        district.setName("District 1");
-        district.setRegionalName("District 1");
-        district.setLanguage(papiamento);
-        district.setCode(1L);
-
-        State state = new State();
-        state.setName("State 1");
-        state.setCode(1L);
-        state.getDistricts().add(district);
-
-        stateDataService.create(state);
-
-        deployedServiceDataService.create(new DeployedService(state, Service.MOBILE_ACADEMY));
-        deployedServiceDataService.create(new DeployedService(state, Service.MOBILE_KUNJI));
-
-        Circle circle = new Circle("AA");
-        circle.setDefaultLanguage(papiamento);
+        final Circle circle = new Circle("AA");
+        circle.setDefaultLanguage(rh.hindiLanguage());
         circleDataService.create(circle);
 
-        Language hi = languageDataService.create(new Language("88", "hindi"));
-
-        District district2 = new District();
-        district2.setName("District 2");
-        district2.setRegionalName("District 2");
-        district2.setLanguage(hi);
-        district2.setCode(2L);
-
-        State state2 = new State();
-        state2.setName("State 2");
-        state2.setCode(2L);
-        state2.getDistricts().add(district2);
+        // Calling these will make sure the districts exist and will map the districts' language to their state
+        rh.newDelhiDistrict();
+        rh.mysuruDistrict();
 
-        stateDataService.create(state2);
+        //TODO: remove this when https://applab.atlassian.net/browse/MOTECH-1679 is fixed
+        circleDataService.doInTransaction(new TransactionCallbackWithoutResult() {
+            @Override
+            protected void doInTransactionWithoutResult(TransactionStatus transactionStatus) {
+                circle.getStates().add(rh.delhiState());
+                circle.getStates().add(rh.karnatakaState());
+                circleDataService.update(circle);
+            }
+        });
```

Comment:
@larubbio @llavoie I was able to resolve the test container failures but hitting a new test failure here - http://ci.motechproject.org/job/NMS/369/testReport/org.motechproject.nms.testing.it.region/CircleServiceBundleIT/testMultipleCirclesSingleState/